functions/imagemagick/ImageMagick/Function.cs

```diff
@@ -0,0 +1,171 @@
+            if (data.Bucket is null || data.Name is null)
+            {
+                _logger.LogError("Malformed GCS event.");
+                return;
+            }
+
+            // Construct URI to GCS bucket and file.
+            string gcsUri = $"gs://{data.Bucket}/{data.Name}";
+            _logger.LogInformation("Analyzing {uri}", gcsUri);
+
+            // Perform safe search detection using the Vision API.
+            Image image = Image.FromUri(gcsUri);
+            SafeSearchAnnotation annotation;
+            try
+            {
+                annotation = await _visionClient.DetectSafeSearchAsync(image);
+            }
+            // If the call to the Vision API fails, log the error but let the function complete normally.
+            // If the exceptions weren't caught (and just propagated) the event would be retried.
+            // See the "Best Practices" section in the documentation for more details about retry.
+            catch (AnnotateImageException e)
+            {
+                _logger.LogError(e, "Vision API reported an error while performing safe search detection");
+                return;
+            }
+            catch (RpcException e)
+            {
+                _logger.LogError(e, "Error communicating with the Vision API");
+                return;
+            }
+
+            if (annotation.Adult == Likelihood.VeryLikely || annotation.Violence == Likelihood.VeryLikely)
+            {
+                _logger.LogInformation("Detected {uri} as inappropriate.", gcsUri);
+                await BlurImageAsync(data, cancellationToken);
+            }
+            else
+            {
+                _logger.LogInformation("Detected {uri} as OK.", gcsUri);
+            }
+        }
+        // [END functions_imagemagick_analyze]
+
+        // [START functions_imagemagick_blur]
+        /// <summary>
+        /// Downloads the Storage object specified by <paramref name="data"/>,
+        /// blurs it using ImageMagick, and uploads it to the "blurred" bucket.
+        /// </summary>
+        private async Task BlurImageAsync(StorageObjectData data, CancellationToken cancellationToken)
+        {
+            // Download image
+            string originalImageFile = Path.GetTempFileName();
+            using (Stream output = File.Create(originalImageFile))
+            {
+                await _storageClient.DownloadObjectAsync(data.Bucket, data.Name, output, cancellationToken: cancellationToken);
+            }
+
+            // Construct the ImageMagick command
+            string blurredImageFile = Path.GetTempFileName();
+            // Command-line arguments for ImageMagick.
+            // Paths are wrapped in quotes in case they contain spaces.
+            string arguments = $"\"{originalImageFile}\" -blur 0x8, \"{blurredImageFile}\"";
+            ProcessStartInfo psi = new ProcessStartInfo("convert", arguments);
+
+            Process process = Process.Start("convert", arguments);
```

Comment:
ImageMagick command is called `convert`? If that's explained somewhere in the docs, that's fine, but it's really confusing otherwise, or for people who just see the code samples.

Reply:
It's in the docs, but I'll add a comment.